internal/appsec/dyngo/operation.go

```diff
@@ -164,13 +161,35 @@ func FromContext(ctx context.Context) (Operation, bool) {
 	return op, ok
 }
 
+// FindOperation looks into the current operation tree for the first operation matching the given type.
+// It has a hardcoded limit of 32 levels of depth even looking for the operation in the parent tree
+func FindOperation[T any, O interface {
+	Operation
+	*T
+}](ctx context.Context) (*T, bool) {
+	op, found := FromContext(ctx)
+	if !found {
+		return nil, false
+	}
+
+	const maxUnwrapping = 32
+	for i := 0; i < maxUnwrapping && op != nil; i++ {
```

Comment:
That feels a little arbitrary, and is new (I don't think we do this for anything else here?)

What is the rationale for limiting here? Do we have indications it would be necessary?